lib/util.js

```diff
@@ -443,23 +456,55 @@ function _fetch_rss_page (feed, page, items, all_items, max_items, oldest_date)
 
         // If we've run into articles which are too old, or we've hit the max
         // number of items, cease crawling
-        const done_crawling = (limited_items.length < feed_json.items.length);
+        const done_crawling = (feed_json.items.length === 0 ||
+                               limited_items.length < feed_json.items.length);
+
+        const new_all_items = new Set(function* () { yield* all_items; yield* new_items.map(item => item.link); }());
+
+        if ((typeof feed_obj !== 'function') && _is_wordpress_feed(feed_json)) {
+            logger.debug('Wordpress feed found, adding pagination');
+            is_paginated = true;
+            const new_feed = _do_create_wordpress(feed_obj);
+            if (Array.isArray(feed)) {
+                feed.shift();
+                feed = [new_feed].concat(feed);
+            } else {
+                feed = new_feed;
+            }
+        }
 
-        if ((is_paginated || is_array) && !done_crawling) {
-            const new_all_items = new Set(function* () { yield* all_items; yield* new_items.map(item => item.link); }());
+        const _continue = (done_crawling && is_paginated && Array.isArray(feed)) ||
+              (!done_crawling && (is_paginated || Array.isArray(feed))) ||
+              (done_crawling && !is_paginated && Array.isArray(feed));
```

Comment:
The first and third clauses are the same, except that `is_paginated` can have either value. So the value of `is_paginated` does not matter to the overall conditional, and you could remove it from both the first and third clauses, and then remove either copy of the clause too.